subsys/bluetooth/mesh/access.c

```diff
@@ -276,11 +282,6 @@ static int metadata_add_model(const struct bt_mesh_model *mod,
 		return 0;
 	}
 
-	if (net_buf_simple_tailroom(buf) < (model_size + BT_MESH_MIC_SHORT)) {
-		LOG_DBG("Model metadata didn't fit in the buffer");
-		return -E2BIG;
-	}
-
```

Comment:
The check did: "If the entire entry can't fit, then return error here". But upon reading the spec, we are supposed to add as many bytes from the entry that can fit, not cutting off at entry boundaries. The `data_buf_...` APIs have been changed, and will now write as many bytes as can fit if an entire write can't fit (see line 174 which is new).

Reply:
Ahh I see, this solves a lot of my other comments. Though, should the user be informed if only parts of the entry is added?

Reply:
I don't see the need. They know there is more data because the can call the API to check the full size and realize that their buffer is smaller and necessarily will contain only the first N bytes. These APIs are quite tailored to the needs of the server models per spec.

Reply:
Remember that these APIs are all internal and not exposed in the public `access.h`.

Reply:
I think it would be good have it documented somewhere (without having to look it up in spec) that the data you pull is partial. It could either be through `LOG_WRN`s or adding a note to the relevant functions that are exposed in `access.h`. 

From a user-perspective, it isn't obvious that it should allow me to pull partial data.

Reply:
"User" here being us in the future, right? I see your point though, I can add a note to the header, but I don't want to do logging here, since this is expected behavior and would happen every single time data is requested if the page is bigger than the buffer (typically the max message buffer)